### PR TITLE
Predictive back migration guide confirmation dialog example

### PR DESCRIPTION
**Pull request by justinmc**
A user requested this example be added to the migration guide [on Discord](https://discord.com/channels/608014603317936148/969301283825659914/1140422179511603290).  I think it's a reasonable request.

Follow up on: https://github.com/flutter/website/pull/8952
Framework PR: https://github.com/flutter/flutter/pull/132249

### CODE REVIEW

File: src/release/breaking-changes/android-predictive-back.md [hunk 1]
```diff
@@ -270,6 +270,41 @@ if (_route.popDisposition == RoutePopDisposition.doNotPop) {
...
+      navigator.pop();
+    }
+  },
+  child: child,
+)
+```
+
 ## Timeline
 
 Landed in version: 3.13<br>
```
Comment (sfshaza2) [inline src/release/breaking-changes/android-predictive-back.md:310]: Can you give me an accurate version for landing in main/beta/pre. I'll take any.  :)

Comment (justinmc) [reply src/release/breaking-changes/android-predictive-back.md:311]: I'm waiting to reland the PR here: https://github.com/flutter/flutter/pull/132249

But once that is merged, I'll update the version number in this PR.  Actually I'll make this PR a draft while I wait for that.

**Comment by justinmc** [comment]
I also decided to add a quick section about supporting predictive back, inspired by https://github.com/flutter/flutter/issues/109513#issuecomment-1666220080.

**Review by parlough — APPROVED** [review]
Thank you for adding this extra example! Looks good to me :)

**Pull request merged**